sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2585,9 +2586,12 @@ class Dataset[T] private[sql](
    * Creates a local temporary view using the given name. The lifetime of this
    * temporary view is tied to the [[SparkSession]] that was used to create this Dataset.
    *
+   * @throws AnalysisException if the view name is invalid
+   *
    * @group basic
    * @since 2.0.0
    */
+  @throws[AnalysisException]
```

Comment:
Then, just document and don't add this annotation?

Actually, I added this because I indeed felt uncomfortable with `public` API raises undocumented exception.
In fact, both `Table already exists` and `Invalid table name` are equally `AnalysisException`.